containers/images/pulp/container-assets/readyz.py

```diff
@@ -3,22 +3,84 @@
 import os
 import sys
 import requests
+import django
+import socket
+from datetime import timedelta
 
-STATUS_PATH = sys.argv[1]
-print(f"Readiness probe checking {STATUS_PATH}")
-response = requests.get(f"http://localhost:24817{STATUS_PATH}", allow_redirects=True)
-data = response.json()
 
-if not data["online_workers"]:
-    sys.exit(1)
+def is_api_health():
+    """
+    Checks if API is health
+    """
+    STATUS_PATH = sys.argv[1]
+    print(f"Readiness probe checking {STATUS_PATH}")
+    response = requests.get(
+        f"http://localhost:24817{STATUS_PATH}", allow_redirects=True
+    )
+    data = response.json()
 
-if not data["online_content_apps"]:
-    sys.exit(2)
+    if not data["online_workers"]:
+        sys.exit(1)
 
-if not data["database_connection"]["connected"]:
-    sys.exit(3)
+    if not data["online_content_apps"]:
+        sys.exit(2)
 
-if os.getenv("REDIS_SERVICE_HOST") and not data["redis_connection"]["connected"]:
-    sys.exit(4)
+    if not data["database_connection"]["connected"]:
+        sys.exit(3)
 
-sys.exit(0)
+    if os.getenv("REDIS_SERVICE_HOST") and not data["redis_connection"]["connected"]:
+        sys.exit(4)
+
+    sys.exit(0)
+
+
+def is_content_health():
+    """
+    Checks if Content is health
+    """
+    print(f"Readiness probe checking {socket.gethostname()}")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "pulpcore.app.settings")
+    django.setup()
+    from django.utils import timezone
+    from pulpcore.app.models.status import ContentAppStatus
+
+    age_threshold = timezone.now() - timedelta(seconds=30)
+    count = ContentAppStatus.objects.filter(
+        name__endswith=socket.gethostname(), last_heartbeat__gte=age_threshold
+    ).count()
+
+    if not count:
+        sys.exit(1)
+
+    sys.exit(0)
+
+
+def is_worker_health():
+    """
+    Checks if Worker is health
+    """
+    print(f"Readiness probe checking {socket.getfqdn()}")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "pulpcore.app.settings")
+    django.setup()
+    from django.utils import timezone
+    from pulpcore.app.models.task import Worker
+
+    age_threshold = timezone.now() - timedelta(seconds=30)
+    count = Worker.objects.filter(
+        name__endswith=socket.getfqdn(), last_heartbeat__gte=age_threshold
```

Comment:
How does the `last_heartbeat__gte` check work?
I'm trying to understand if it would be good to somehow "sync" the `timedelta(seconds=30)` with the readinessProbe `periodSeconds` (interval between each probe).
For example, if we configure `periodSeconds=10` would the probes get a response from a django heartbeat made 20 seconds ago?

Reply:
this is how `online` is defined on pulpcore:
https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/task.py#L104-L117

Reply:
> For example, if we configure `periodSeconds=10` would the probes get a response from a django heartbeat made 20 seconds ago?

every 10 seconds the query would verify if a heartbeat has happened in the latest 30 seconds, so 20 seconds ago would be included